### PR TITLE
SCUMM: Correctly flag demos as demos in the detection

### PR DESCRIPTION
**Pull request by henke37**
Probably need a tweak or two to the hash database file too. Some help there please.

### CODE REVIEW

**Comment by bluegr** [comment]
Please remove the irrelevant commits, and change the commit messages to confirm to our guidelines

**Comment by digitall** [comment]
@bluegr: I have made the changes you have indicated and filtered out the irrelevant commits, since it appears that this PR is not likely to be cleaned up. This can be found at https://github.com/digitall/scummvm/tree/pr931_fixup. I can either open this as a new PR and/or if it looks fine, rebase / cherry pick this to scummvm/master, though would prefer a bit of review from SCUMM engine devs.

**Comment by henke37** [comment]
Thanks man. I've been so busy that I haven't had time to do what should be trivial changes.

**Comment by digitall** [comment]
@henke37: No problem. Just trying to clean up various pending PRs. :)

**Comment by digitall** [comment]
@sev- : Could you or one of the other SCUMM engine developers review my amended branch at https://github.com/digitall/scummvm/tree/pr931_fixup ? If you are busy, can you suggest another reviewer or should I just commit that revised code as-is?

File: backends/platform/sdl/win32/win32.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ void OSystem_Win32::init() {
 
 void OSystem_Win32::initBackend() {
 	// Console window is enabled by default on Windows
-	ConfMan.registerDefault("console", true);
+	ConfMan.registerDefault("console", false);
```
Comment (sev-) [inline backends/platform/sdl/win32/win32.cpp]: This does not belong here

File: engines/scumm/detection.cpp [hunk 2]
```diff
@@ -489,11 +489,16 @@ static void computeGameSettingsFromMD5(const Common::FSList &fslist, const GameF
...
 	// Compute the precise game settings using gameVariantsTable.
 	for (const GameSettings *g = gameVariantsTable; g->gameid; ++g) {
-		if (g->gameid[0] == 0 || !scumm_stricmp(md5Entry->gameid, g->gameid)) {
+		if (g->gameid[0] != 0 && !!scumm_stricmp(md5Entry->gameid, g->gameid)) continue;
 			// The gameid either matches, or is empty. The latter indicates
 			// a generic entry, currently used for some generic HE settings.
-			if (g->variant == 0 || !scumm_stricmp(md5Entry->variant, g->variant)) {
-				// Perfect match found, use it and stop the loop
+
+		if (g->variant != 0 && !!scumm_stricmp(md5Entry->variant, g->variant)) continue;
```
Comment (sev-) [inline engines/scumm/detection.cpp]: The code formatting in these lines is not correct.

Also, I do not quite understand what is the goal of these.

File: backends/platform/sdl/win32/win32.cpp (hunk 1 above)
Comment (digitall) [reply backends/platform/sdl/win32/win32.cpp]: @sev- : This and the other irrelevant commits have been removed in my refactoring of this PR as:
https://github.com/digitall/scummvm/tree/pr931_fixup

I have also fixed the formatting as per your next commit. I can't force push over this PR AFAIK as I am not a collaborator to @henke37 on this.

I do not know the intention and advantages of this, so would need him to respond, but if not we can close this and create a new PR from my fixup branch if needs be.

**Comment by csnover** [comment]
@digitall Please go ahead and open a new PR with your fixed up branch if this changeset is still desired.

**Comment by digitall** [comment]
@csnover : Not sure, but have opened PR #1043 for the cleaned up commit for the SCUMM engine detection. Hopefully @sev- or another SCUMM engine dev can review this and accept / reject as needs be.

**Pull request closed without merging**